fireant/slicer/queries.py

```diff
@@ -230,9 +232,20 @@ def _build_reference_query(self, query, database, references, table, joins, metr
 
             # The interval term from pypika does not take into account leap years, therefore the interval
             # needs to be replaced with a database specific one when appropriate.
-            if reference_key in [YoY.key, Delta.generate_key(YoY.key), DeltaPercentage.generate_key(YoY.key)]:
+            yoy_keys = [YoY.key, Delta.generate_key(YoY.key), DeltaPercentage.generate_key(YoY.key)]
+            if reference_key in yoy_keys:
                 interval = database.interval(years=1)
 
+                # If YoY reference is used with the week interval, the dates will fail to match in the join
+                # as the first day of the ISO year is different. Therefore, we truncate the reference date by adding
+                # a year, truncating it and removing a year so the dates match.
+                from fireant.slicer import DatetimeDimension
+                from fireant.database import vertica
+                week = DatetimeDimension.week
+                if hasattr(dimensions[dimension_key], 'date_format') \
+                        and dimensions[dimension_key].date_format in [week, vertica.DATETIME_INTERVALS[week].size]:
```

Comment:
I think it is safer to have the DATETIME_INTERVALS constant defined inside VerticaDatabase and then use database.DATETIME_INTERVALS here. As soon as we add another Database, which could have a similar issue, this could be a source of painful bug tracking. If all Database classes were supposed to have it then that would be a no issue.